--- a/src/main.c
+++ b/src/main.c
@@ -411,7 +411,8 @@
 
         g_log_set_default_handler (gsd_log_default_handler, NULL);
 
-        if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME")) {
+        if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME") != 0 ||
+            g_strcmp0(g_getenv("DESKTOP_SESSION"), "gnome") != 0) {
                 xsync = g_slice_new0 (MetaXSync);
                 init_xsync();
                 mainloop = g_main_loop_new (NULL, FALSE);
